lib/models/builder.js

```diff
@@ -44,9 +44,14 @@ class Builder extends CoreObject {
    */
   readBuildFile(path) {
```

Comment:
Let's refactor this function to not take in the path argument here, it is more confusing overall than having the method itself control all of the logic (especially when the thing we get isn't the path to the file, its a directory path).